src/server/pps/server/api_server.go

```diff
@@ -2713,7 +2713,7 @@ func (a *apiServer) deletePipeline(pachClient *client.APIClient, request *pps.De
 
 	// check if the output repo exists--if not, the pipeline is non-functional and
 	// the rest of the delete operation continues without any auth checks
-	if _, err := pachClient.InspectRepo(request.Pipeline.Name); err != nil && !isNotFoundErr(err) {
+	if _, err := pachClient.InspectRepo(request.Pipeline.Name); err != nil && !isNotFoundErr(err) && !auth.IsErrNoRoleBinding(err) {
 		return nil, err
 	} else if !isNotFoundErr(err) {
```

Comment:
Should this check be weakened too? I.e. `else if !isNotFoundErr(err) && !auth.IsErrNoRoleBinding(err)` or even `else if err == nil`?

If I understand this correctly, it calls `DeleteRepo` in the `IsErrNoRoleBinding(err) == true` case, which I think it's not supposed to do?

Reply:
Yup, good catch.